src/cli/commands/upgrade.js

```diff
@@ -27,19 +29,43 @@ export async function run(
     peerDependencies,
   } = await config.readRootManifest() || {};
   const allDependencies = Object.assign({}, peerDependencies, optionalDependencies, devDependencies, dependencies);
+  let addArgs = [];
 
-  const addArgs = args.map((dependency) => {
-    const remoteSource = allDependencies[dependency];
+  if (flags.scope) {
+    flags.scope = flags.scope.replace(/\/?$/g, '/');
+    flags.scope = flags.scope.replace(/^@?/g, '@');
 
-    if (remoteSource && PackageRequest.getExoticResolver(remoteSource)) {
-      return remoteSource;
-    }
+    if (/^@\S*\/$/g.test(flags.scope)) {
```

Comment:
Use `[a-zA-Z0-9-][a-zA-Z0-9_.-]*` instead of `\S*`, this way we ensure that the scope is a valid scope according to npm rules (and we won't risk any schenanigans with extra `/`).